### PR TITLE
Kubernetes - Prometheus Rule File Generator

### PR DESCRIPTION
**Pull request by umangachapagain**
Build method that provides Prometheus Rule file which can be applied to K8s or OC cluster directly.

Issue: gluster/gluster-mixins/issues/25
Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

### CODE REVIEW

**Comment by cloudbehl** [comment]
Looks great :+1: 

Can you also add README
1. To install from JB
2. to generate manifest
3. To apply manifest

Also, The Grafana part is still missing we need to do same for grafana also.

**Comment by umangachapagain** [comment]
> Looks great +1
> 
> Can you also add README
> 
>     1. To install from JB
> 
>     2. install jsontoyaml(go get github.com/brancz/gojsontoyaml)
> 
>     3. to generate manifest
> 
>     4. To apply manifest 
> 
Done.
> 
> Also, The Grafana part is still missing we need to do same for grafana also.
>
I am fairly new to Grafana Dashboards and Configurations. Will need some time to grasp the flow.
Also, need some functional dashboards and configs to be checked in to write deployment.

File: extras/manifests/prometheus-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: k8s
```
Comment (cloudbehl) [inline extras/manifests/prometheus-rules.yaml]: name of Prometheus is alert https://github.com/gluster/gcs/blob/29531c63d6c0e602dffee0a744084b2562da536b/deploy/templates/gcs-manifests/gcs-prometheus-bundle.yml.j2#L79

**Comment by cloudbehl** [comment]
> I am fairly new to Grafana Dashboards and Configurations. Will need some time to grasp the flow.
> Also, need some functional dashboards and configs to be checked in to write deployment.

@umangachapagain No problem, let's target this PR only for the Prometheus related stuff.

File: extras/manifests/prometheus-rules.yaml (hunk 1 above)
Comment (umangachapagain) [reply extras/manifests/prometheus-rules.yaml]: Ack.
But this yaml contains alerts and recording rules both. Naming it something like `alert` or `rules` is just misleading and makes no sense. 
Instead we should name it `gluster` or something similar. 

This change would have to be made in both `gcs` and `gluster-mixins`. Thoughts?
@cloudbehl @shtripat

**Pull request merged**